### ISSUE TITLE
Build on fedora 36 R 4.1.3

### ISSUE THREAD

**Issue by Patrikios** [arg0naut91/chronshape#28]
While testing the build of version 0.1.4 on Fedora 36, R 4.1.3:
- can install the package just fine, all function-examples are working OK
- devtools::check() runs without errors or warnings however 1 note is present:
```
❯ checking compilation flags used ... NOTE
  Compilation used the following non-portable flag(s):
    ‘-Werror=format-security’ ‘-Wp,-D_FORTIFY_SOURCE=2’
    ‘-Wp,-D_GLIBCXX_ASSERTIONS’
```

- devtools::test() fails - I think because of the further compilation flags

**Comment by Patrikios** [comment]
After several tries, reinstalling, reloading the latest commit 0.1.4, tests are now run. Maybe the note above should be dealt with in order to remove potential Fedora build errors.

**Comment by arg0naut91** [comment]
If the package installs fine and all checks/tests run OK on Fedora (apart from the NOTE and at least during the CMD check itself, e.g. via devtools as you describe), I think the NOTE mentioned is unlikely to be an issue (within the actual build as well as for the further submissions to CRAN). See also similar GH discussions ([here](https://github.com/glotaran/TIMP/issues/3) and [here](https://github.com/r-hub/rhub/issues/264)) as well as [this](https://stackoverflow.com/questions/68584588/building-r-package-using-rcpp) SO question and comments. Some even suggest this is a bug within the R check code itself.

Regarding your second comment:

> After several tries, reinstalling, reloading the latest commit 0.1.4, tests are now run. 

Do you mean that the `devtools::test()` runs fine now, or the tests within check? If `devtools::test()` still doesn't run fine for you, it'd be interesting to see the logs (if you can get any) with corresponding warnings/errors.

**Comment by Patrikios** [comment]
First what happenend is that I freshly cloned the  repo and did `test()`. That failed. Then I tried `check()` and there was only a `NOTE `present. Then I wanted to test it it builds, hence I uninstalled the previous version and installed the new, without issue. Again uninstalled and tried to build again, as I wanted to get the exact problem stating from R. And voila , the `test()` worked. As u mentioned, it seems to be a build specific problem. If it will persist in the future, I will try to adjust the Makevars as mentioned in the posts. But for now, closing the issue.
